client/restarts_test.go

```diff
@@ -99,7 +99,7 @@ func TestClient_RestartTracker_RestartTriggered(t *testing.T) {
 	p := testPolicy(true, structs.RestartPolicyModeFail)
 	p.Attempts = 0
 	rt := newRestartTracker(p, structs.JobTypeService)
-	if state, when := rt.SetRestartTriggered().GetState(); state != structs.TaskRestarting && when != 0 {
+	if state, when := rt.SetRestartTriggered(false).GetState(); state != structs.TaskRestarting && when != 0 {
```

Comment:
Unit test the new case

Reply:
Added https://github.com/hashicorp/nomad/pull/3105/commits/cde908e35d8b3dedc2ed68d2d7b2eb832aaf06ad